isb_lib/core.py

```diff
@@ -40,8 +48,89 @@ def relationAsSolrDoc(
+    if _shouldAddMetadataValueToSolrDoc(coreMetadata, "keywords"):
+        doc["keywords"] = coreMetadata["keywords"]
+    if _shouldAddMetadataValueToSolrDoc(coreMetadata, "registrant"):
+        doc["registrant"] = coreMetadata["registrant"]
+    if _shouldAddMetadataValueToSolrDoc(coreMetadata, "samplingPurpose"):
+        doc["samplingPurpose"] = coreMetadata["samplingPurpose"]
+
+    if "producedBy" in coreMetadata:
+        # The solr index flattens subdictionaries, so check the keys explicitly in the subdictionary to see if they should be added to the index
+        producedBy = coreMetadata["producedBy"]
+        if _shouldAddMetadataValueToSolrDoc(producedBy, "label"):
+            doc["producedBy_label"] = producedBy["label"]
+        if _shouldAddMetadataValueToSolrDoc(producedBy, "description"):
+            doc["producedBy_description"] = producedBy["description"]
+        if _shouldAddMetadataValueToSolrDoc(producedBy, "responsibility"):
+            doc["producedBy_responsibility"] = producedBy["responsibility"]
+        if _shouldAddMetadataValueToSolrDoc(producedBy, "hasFeatureOfInterest"):
+            doc["producedBy_hasFeatureOfInterest"] = producedBy["hasFeatureOfInterest"]
+        if _shouldAddMetadataValueToSolrDoc(producedBy, "resultTime"):
+            raw_date_str = producedBy["resultTime"]
+            # TODO: https://github.com/isamplesorg/isamples_inabox/issues/24
+            date_time = dateparser.parse(raw_date_str)
+            if date_time is not None:
+                doc["producedBy_resultTime"] = datetimeToSolrStr(date_time)
+
+        if "samplingSite" in producedBy:
+            samplingSite = producedBy["samplingSite"]
+            if _shouldAddMetadataValueToSolrDoc(samplingSite, "description"):
+                doc["producedBy_samplingSite_description"] = samplingSite["description"]
+            if _shouldAddMetadataValueToSolrDoc(samplingSite, "label"):
+                doc["producedBy_samplingSite_label"] = samplingSite["label"]
+            if _shouldAddMetadataValueToSolrDoc(samplingSite, "placeName"):
+                doc["producedBy_samplingSite_placeName"] = samplingSite["placeName"]
+
+            if "location" in samplingSite:
+                location = samplingSite["location"]
+                if _shouldAddMetadataValueToSolrDoc(location, "elevation"):
+                    location_str = location["elevation"]
+                    match = ELEVATION_PATTERN.match(location_str)
+                    if match is not None:
+                        doc["producedBy_samplingSite_location_elevation"] = float(match.group(1))
+                if _shouldAddMetadataValueToSolrDoc(location, "latitude") and _shouldAddMetadataValueToSolrDoc(location, "longitude"):
+                    doc["producedBy_samplingSite_location_latlon"] = f"{location['latitude']},{location['longitude']}"
+
+
+    return doc
+
 
-def solrAddRecords(rsession, records, url="http://localhost:8983/solr/isb_rel/"):
```

Comment:
I removed this because I ran into a couple cases where I forgot to include it and it was trying to run modifications against `isb_rel` for the core records.  Could probably modularize it more than just forcing it onto callers, but I haven't thought about how yet.

Reply:
That's fine. Better to force a choice when there's multiple options